src/datacube.ts

```diff
@@ -43,6 +43,26 @@ export class DataCube {
       new DataSet(this.endpoint, { dataSetIri, dataSetLabel, graphIri }));
   }
 
+  /**
+   * Fetch a [[DataSet]] by its IRI.
+   *
+   * @param iri IRI of the DataSet to return.
+   */
+  public async datasetByIri(iri: string): Promise<DataSet> {
```

Comment:
Shouldn't this type be `Promise<DataSet | undefined>` since `.find()` is not guaranteed to return a result?